dev-util/bpftool/bpftool-6.10.ebuild

```diff
@@ -0,0 +1,125 @@
+HOMEPAGE="https://kernel.org/"
+
+LINUX_V="${PV:0:1}.x"
+LINUX_VER=$(ver_cut 1-2)
+LINUX_PATCH=patch-${PV}.xz
+SRC_URI="https://www.kernel.org/pub/linux/kernel/v${LINUX_V}/${LINUX_PATCH}"
+
+LINUX_SOURCES="linux-${LINUX_VER}.tar.xz"
+SRC_URI+=" https://www.kernel.org/pub/linux/kernel/v${LINUX_V}/${LINUX_SOURCES}"
+
+S_K="${WORKDIR}/linux-${LINUX_VER}"
+S="${S_K}/tools/bpf/bpftool"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc ~ppc64 ~riscv ~x86"
+IUSE="caps +llvm"
+
+RDEPEND="
+	sys-libs/binutils-libs:=
+	sys-libs/zlib:=
+	virtual/libelf:=
+	caps? ( sys-libs/libcap:= )
+	llvm? ( sys-devel/llvm:= )
+"
+DEPEND="
+	${RDEPEND}
+	>=sys-kernel/linux-headers-5.8
+"
+BDEPEND="
+	${LINUX_PATCH+dev-util/patchutils}
+	${PYTHON_DEPS}
+	app-arch/tar
+	dev-python/docutils
+"
+
+CONFIG_CHECK="~DEBUG_INFO_BTF"
+
+# src_unpack and src_prepare are copied from dev-util/perf since
+# it's building from the same tarball, please keep it in sync with perf
+src_unpack() {
+	local paths=(
+		kernel/bpf tools/{arch,bpf,build,include,lib,perf,scripts}
+		scripts include lib "arch/*/include" "arch/*/lib" "arch/*/tools"
+	)
+
+	# We expect the tar implementation to support the -j and --wildcards option
+	echo ">>> Unpacking ${LINUX_SOURCES} (${paths[*]}) to ${PWD}"
+	gtar --wildcards -xpf "${DISTDIR}"/${LINUX_SOURCES} \
+		"${paths[@]/#/linux-${LINUX_VER}/}" || die
+
+	if [[ -n ${LINUX_PATCH} ]] ; then
+		eshopts_push -o noglob
+		ebegin "Filtering partial source patch"
+		filterdiff -p1 ${paths[@]/#/-i } -z "${DISTDIR}"/${LINUX_PATCH} \
+			> ${P}.patch
+		eend $? || die "filterdiff failed"
+		eshopts_pop
+	fi
+
+	local a
+	for a in ${A}; do
+		[[ ${a} == ${LINUX_SOURCES} ]] && continue
+		[[ ${a} == ${LINUX_PATCH} ]] && continue
+		unpack ${a}
+	done
+}
+
+src_prepare() {
+	default
+
+	if [[ -n ${LINUX_PATCH} ]] ; then
+		pushd "${S_K}" >/dev/null || die
+		eapply "${WORKDIR}"/${P}.patch
+		popd || die
+	fi
+
+	# Use rst2man or rst2man.py depending on which one exists (#930076)
+	type -P rst2man >/dev/null || sed -i -e 's/rst2man/rst2man.py/g' Documentation/Makefile || die
+
+	# remove -Werror (bug 887981)
+	sed -i -e 's/\-Werror//g' ../../lib/bpf/Makefile || die
+}
+
+bpftool_make() {
+	local arch=$(tc-arch-kernel)
+	tc-export AR CC LD
+
+	emake V=1 VF=1 \
+		HOSTCC="$(tc-getBUILD_CC)" HOSTLD="$(tc-getBUILD_LD)" \
+		EXTRA_CFLAGS="${CFLAGS}" ARCH="${arch}" \
+		prefix="${EPREFIX}"/usr \
+		bash_compdir="$(get_bashcompdir)" \
+		feature-libcap="$(usex caps 1 0)" \
+		feature-llvm="$(usex llvm 1 0)" \
+		"$@"
+}
+
+src_compile() {
+	bpftool_make
+	bpftool_make -C Documentation
+}
+
+src_install() {
+	bpftool_make DESTDIR="${D}" install
+	bpftool_make mandir="${ED}"/usr/share/man -C Documentation install
+}
+
+pkg_postinst() {
+	optfeature "clang-bpf-co-re support" sys-devel/clang[llvm_targets_BPF]
+
+	elog "Note that 'bpftool --version' output differs from the ebuild version."
```

Comment:
What do you think about naming the ebuild to match the output? In this case it would be neccesarry to manually update kernel versioning variables which are used to pull the tarballs, is it too much of a hassle in your opinion?

Reply:
I thought about that and tried it. It would make the kernel dependency much easier (just a hardcoded number), but it would not really improve anything because the printed version has no correlation to bpftool changes; it's more a version of the built-in libbpf. So you could have kernel x printing 7.x.0 and kernel x+1 printing the same, despite the fact that the bpftool code changed but not the libbpf version.
Checkout the bpftool commit in the linked bug; this is all really dumb and I don't like it either, but using the kernel number for the ebuild mitigates this problem at least a bit. 
I would even prefer to keep the ebuild version and the oputput in sync to reduce confusion, but how would we then indicate small version increments? revbumps? using the kernel version as slot?

Reply:
I see, that's really messy for packaging. Perhaps instead of pushing an update with every new kernel release we could just follow when releases are being cut in the [bpftool github repo](https://github.com/libbpf/bpftool) and use the corresponding kernel tarball, but this would mean less frequent updates.

Reply:
I made an alternative version in https://github.com/hhoffstaette/gentoo/commit/c898fa18664abf90a3981e88211eca242ecf7091 - basically just version the build the same as the --version output and use PV to indicate which tarball to get. I don't expect frequent updates but in any case we can always just use revbumps for minor updates.
If there are no other objections I'll amend this PR with that change.

Reply:
Sounds good to me

Reply:
pushed!